### PR TITLE
docs: minor changes to dev docs

### LINKED ISSUE #13164: docs: updates to contribution guidelines
### The problem

Making small changes to contribution guidelines so docs are a bit clearer

### The solution

updates to guidelines

### Examples

_No response_

### Application/PAL

_No response_

### Business priority

None

### Available extra resources

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #13164 

#### Testing / Reviewing

- Open to any suggestions to making this clearer. Was trying to think of these small changes from onboarding and contribution perspective

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -93,7 +95,28 @@ upstream    git@github.com:carbon-design-system/carbon.git (fetch)
...
 
-### 4. Work in a branch
+Great job on getting to this point! Now it's time to code :computer:
+
+## - Setting up your code -
+
+### 1. Find or create an issue
+
+Before you start coding, if you have an issue in mind that you'd like to tackle,
+please first take a [search](https://github.com/IBM/carbon-components/issues)
```
Comment (alisonjoseph) [inline .github/CONTRIBUTING.md]:
```suggestion
please first [search](https://github.com/IBM/carbon-components/issues)
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -93,7 +95,28 @@ upstream    git@github.com:carbon-design-system/carbon.git (fetch)
...
+the issues to see if a similar issue you'd like to work on has already been
+submitted. If a similar issue has been submitted,assign yourself or ask to be
```
Comment (alisonjoseph) [inline .github/CONTRIBUTING.md]:
```suggestion
submitted. If a similar issue has been submitted, assign yourself or ask to be
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -93,7 +95,28 @@ upstream    git@github.com:carbon-design-system/carbon.git (fetch)
...
+assigned to the issue by posting a comment.
+
+If the issue does not exist, please make a new issue. Issues give us context
+about what you are contributing and expedite the process to getting your
+contributions merged into Carbon. It's a win for everybody :tada:
+
+If you have no idea what you'd like to contribute, take a look at our backlog of
+issues and take your pick! Our issues with the label
+[`good first issue`](https://github.com/carbon-design-system/carbon/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue+%F0%9F%91%8B%22)
+are also a great starting point. 👋
```
Comment (alisonjoseph) [inline .github/CONTRIBUTING.md]:
```suggestion
are a great starting point. 👋
```

**Review by alisonjoseph — COMMENTED** [review]
Thanks for updating this! added a couple small suggestions

**Review by tw15egan — APPROVED** [review]
Looks great pending Alisons suggestions 👍🏻 ✅

**Comment by andreancardona** [comment]
> Thanks for updating this! added a couple small suggestions

awesome, just made all those changes!

**Review by alisonjoseph — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Comment by tay1orjones** [comment]
@andreancardona I think this just needs a `yarn format`

**Pull request merged**